server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourceContext.java

```diff
@@ -329,17 +355,17 @@ public boolean equals(Object o) {
         if (fetchSource != that.fetchSource) return false;
         if (Arrays.equals(excludes, that.excludes) == false) return false;
         if (Arrays.equals(includes, that.includes) == false) return false;
-        if (includeVectors != that.includeVectors) return false;
-
-        return true;
+        return includeVectors == that.includeVectors;
```

Comment:
Would rather keep the `return true` at the end of this to keep the structure the same and friendlier to future updates

Reply:
Ahh, My mistake. I wanted to check what IntellJ was suggesting but never reverted back.